sycl/test-e2e/ClusterLaunch/cluster_launch_enqueue_functions.cpp

```diff
@@ -0,0 +1,59 @@
+// REQUIRES: cuda
+// RUN: %{build} -Xsycl-target-backend --cuda-gpu-arch=sm_90 -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/ext/oneapi/experimental/cluster_group_prop.hpp>
+#include <sycl/ext/oneapi/experimental/enqueue_functions.hpp>
+#include <sycl/sycl.hpp>
+
+#include <string>
+
+int main() {
+  using namespace sycl::ext::oneapi::experimental;
+
+  sycl::queue queue;
+  auto computeCapability = std::stof(
+      queue.get_device().get_info<sycl::info::device::backend_version>());
+
+  if (computeCapability >= 9.0) {
+    cuda::cluster_size cluster_dims(sycl::range<3>(2, 2, 1));
+    properties cluster_launch_property{cluster_dims};
+
+    int *correct_result_flag = sycl::malloc_device<int>(1, queue);
+    queue.memset(correct_result_flag, 1, sizeof(int)).wait();
+
+    submit_with_event(queue, [&](sycl::handler &cgh) {
+      nd_launch(cgh,
+                launch_config(sycl::nd_range<3>({64, 64, 1}, {32, 32, 1}),
+                              cluster_launch_property),
+                [=](sycl::nd_item<3> it) {
+                  uint32_t cluster_dim_x, cluster_dim_y, cluster_dim_z;
+// Temporary solution till cluster group class is implemented
+#if defined(__SYCL_DEVICE_ONLY__) && defined(__SYCL_CUDA_ARCH__) &&            \
+    (__SYCL_CUDA_ARCH__ >= 900)
+                  asm volatile("\n\t"
+                               "mov.u32 %0, %%cluster_nctaid.x; \n\t"
+                               "mov.u32 %1, %%cluster_nctaid.y; \n\t"
+                               "mov.u32 %2, %%cluster_nctaid.z; \n\t"
+                               : "=r"(cluster_dim_z), "=r"(cluster_dim_y),
```

Comment:
> Should we also be testing for is_explicit_cluster,``clusterid, nclusterid` , `cluster_ctaid`, `cluster_ctarank` `cluster_nctarank`?

I would suppose these will come with the full implementation of the cluster group spec, and test it in some fashion, because those sregs are more related to the id query of different relevant information associated with the clusters, the clusters id in the kernel, id of workgroup within the cluster etc...

This partial implementation focuses on the cluster launch, and checks whether or not the cluster range (as in the number of wgs in the cluster) are being correctly reflected or not, hence the only test

Reply:
OK sure